_blog/2019/asylo-redis-sqlite.md

```diff
@@ -0,0 +1,190 @@
+trusted, and it opens up the door for exciting new types of systems.
+
+Being isolated from the operating system and peripheral hardware (e.g., disk
+storage, network cards, etc), however, means that an enclave application doesn’t
+have direct access to the functionality that an operating system or special
+hardware would provide. While some applications wouldn’t require that kind of
+additional functionality, the vast majority do require it to varying degrees.
+
+Asylo works to bridge this gap by providing the same POSIX interfaces to this
+functionality that applications would invoke on a non-enclave system. However,
+the Asylo versions of these interfaces work behind the scenes to interact with
+the untrusted operating system in ways that don’t trust it. This can involve
+encrypting data, tokenizing potentially sensitive values, or verifying that the
+operating system is not being malicious before using its responses.
+
+One example of the POSIX support is signals. Many real-world applications
+register and handle signals for various purposes. The enclaves do not support
+signal handling. Signals are sent by the kernel to the host operating system.
+Asylo provides a secure signal handling system that allows incoming signals to
+enter and notify the enclave, which invokes handlers registered inside the
+enclave.
+
+Another example is fork(). This is also widely used by real-world applications.
+For security reasons, enclaves are not cloneable. If fork() is called on the
+host operating system with a running enclave, the enclave will not exist in the
+new process (the address space for the enclave will simply be empty). Asylo
+provides secure fork() that safely encrypts a snapshot of the parent enclave and
+restores it in the child enclave.
+
+As Asylo continues to add more and more POSIX interfaces in this manner, it
+becomes easier and easier to port applications to run in enclaves. This blog
+post discusses the technical details behind some interesting new additions
+(including signals and fork) that significantly improve this. In fact, as of
+Asylo 0.4.0, one can run a wide variety of real-world applications (including
+Redis and SQLite) without even modifying any of their source code!
+
+## Technical Details
+In recent months, several enhancements to the Asylo framework enabled the
+enclavization of non-trivial user applications such as Redis and SQLite. These
+enhancements include the addition of the `cc_enclave_binary` rule that enables
+wrapping of whole C++ applications as well as additional POSIX support. We
+briefly describe some of the highlights below.
+
+### Application wrapper
+
+To make it easy to run an application in an enclave, Asylo now provides an
+application wrapper, which enables users to easily run an existing application
+in an enclave without any source code modifications.
+
+Asylo now provides the [`cc_enclave_binary` Bazel macro](https://github.com/google/asylo/blob/master/asylo/bazel/asylo.bzl#L457)
```

Comment:
This link won't point to cc_enclave_binary any more if we modify asylo.bzl; please point directly to 0.4.0 instead of master.

Reply:
Fixed in the new commit